Source/WebCore/platform/graphics/GraphicsContext.h

```diff
@@ -246,10 +246,11 @@ class GraphicsContext {
     WEBCORE_EXPORT virtual RefPtr<ImageBuffer> createAlignedImageBuffer(const FloatSize&, const DestinationColorSpace& = DestinationColorSpace::SRGB(), std::optional<RenderingMethod> = std::nullopt) const;
     WEBCORE_EXPORT virtual RefPtr<ImageBuffer> createAlignedImageBuffer(const FloatRect&, const DestinationColorSpace& = DestinationColorSpace::SRGB(), std::optional<RenderingMethod> = std::nullopt) const;
 
-    virtual void drawNativeImage(NativeImage&, const FloatSize& selfSize, const FloatRect& destRect, const FloatRect& srcRect, const ImagePaintingOptions& = { }) = 0;
-
     virtual bool needsCachedNativeImageInvalidationWorkaround(RenderingMode) { return true; }
 
+    WEBCORE_EXPORT void drawNativeImage(NativeImage&, const FloatSize& selfSize, const FloatRect& destRect, const FloatRect& srcRect, const ImagePaintingOptions& = { });
```

Comment:
As a read of the header it's hard for me to know what the difference is between `drawNativeImage` and `drawPlatformImage`. This needs a comment, or some refactoring that makes it impossible to call the wrong thing from outside GraphicsContext.

Reply:
e.g. should `drawPlatformImage` be private, with `NativeImage` being a friend class?

Reply:
Rather than have both "platform" and "native", maybe keep the existing `drawNativeImage()` and add `drawNativeImageInternal()` or something.

Reply:
I added the virtual method `drawNativeImageInternal()` and I made the following classes be friends of `GraphicsContext`:

1. BifurcatedGraphicsContext;
2. DisplayList::DrawNativeImage;
3. NativeImage;